crates/rooch/src/commands/account/commands/sign.rs

```diff
@@ -5,57 +5,73 @@ use crate::cli_types::{CommandAction, WalletContextOptions};
 use async_trait::async_trait;
 use clap::Parser;
 use hex::ToHex;
+use moveos_types::state::MoveState;
 use rooch_key::keystore::account_keystore::AccountKeystore;
-use rooch_rpc_api::jsonrpc_types::account_sign_view::AccountSignView;
 use rooch_types::{
     address::ParsedAddress,
-    error::{RoochError, RoochResult},
+    error::RoochResult,
+    framework::auth_payload::{SignData, MESSAGE_INFO, MESSAGE_INFO_PREFIX},
 };
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
 
-/// Sign an msg with current account private key (sign_hashed)
-///
-/// This operation must be specified with -a or
-/// --address to export only one address with a private key.
+/// Sign a message with a parsed address
 #[derive(Debug, Parser)]
 pub struct SignCommand {
+    // An address to be used
     #[clap(short = 'a', long = "address", value_parser=ParsedAddress::parse, default_value = "")]
     address: ParsedAddress,
+
+    /// A message to be signed
+    #[clap(short = 'm', long)]
+    message: String,
+
     #[clap(flatten)]
     pub context_options: WalletContextOptions,
 
     /// Return command outputs in json format
     #[clap(long, default_value = "false")]
     json: bool,
+}
 
-    /// Msg command will sign
-    #[clap(long, default_value = "")]
-    msg: String,
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct SignAccountOutput {
+    pub signature_hash: String,
+    pub message_hash: String,
 }
 
 #[async_trait]
-impl CommandAction<Option<AccountSignView>> for SignCommand {
-    async fn execute(self) -> RoochResult<Option<AccountSignView>> {
+impl CommandAction<Option<SignAccountOutput>> for SignCommand {
+    async fn execute(self) -> RoochResult<Option<SignAccountOutput>> {
         let context = self.context_options.build_require_password()?;
         let password = context.get_password();
-
         let mapping = context.address_mapping();
-        let addrss = self.address.into_rooch_address(&mapping).map_err(|e| {
-            RoochError::CommandArgumentError(format!("Invalid Rooch address String: {}", e))
-        })?;
+        let rooch_address = self.address.into_rooch_address(&mapping)?;
+
+        let mut message_info = Vec::new();
+        message_info.append(&mut MESSAGE_INFO.to_vec());
```

Comment:
You do not need to include the MESSAGE_INFO const. The MESSAGE_INFO is the rooch transaction sign message, and this custom sign message is from the cli argument.

Reply:
> You do not need to include the MESSAGE_INFO const. The MESSAGE_INFO is the rooch transaction sign message, and this custom sign message is from the cli argument.

Resolved.